packages/esbuild-plugin-node/src/common.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ModuleParams } from './types';
+
+export function wrapModule(
+  originalSource: string,
+  {
+    oTelInstrumentationPackage,
+    oTelInstrumentationClass,
+    instrumentationName,
+    oTelInstrumentationConstructorArgs = '',
+  }: ModuleParams
+) {
+  return `
+(function() {
+  ${originalSource}
+})(...arguments);
+{
```

Comment:
This is the crux of the approach: Wrap the original module contents in an IIFE, then once that has evaluated take the `module.exports` and pass them to the appropriate instrumentation's patch method, then re-export that